### PR TITLE
SONARPY-348 Migrate documentation to be embedded in analyzer

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

**Comment by nicolas-harraudeau-sonarsource** [comment]
@benzonico The new doc is not replacing the default one in SonarQube 8. I guess this is because the new file is in "sonar-python-plugin/src/**test**/resources/static/documentation.md".

**Comment by benzonico** [comment]
d'oh
(fixed)

File: sonar-python-plugin/src/main/resources/static/documentation.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+### Overview
+
+The Python analyzer parses the source code, creates an Abstract Syntax Tree (AST) and then walks through the entire tree. A coding rule is a visitor that is able to visit nodes from this AST.
+
+As soon as the coding rule visits a node, it can navigate its children and log issues if necessary.
+
+### Writing a Plugin
+
+Custom rules for Python can be added by writing a SonarQube Plugin and using Python analyzer APIs.
+Here are the step to follow:
```
Comment (nicolas-harraudeau-sonarsource) [inline sonar-python-plugin/src/main/resources/static/documentation.md]: s/step/steps/

File: sonar-python-plugin/src/main/resources/static/documentation.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+#### Create SonarQube Plugin
```
Comment (nicolas-harraudeau-sonarsource) [inline sonar-python-plugin/src/main/resources/static/documentation.md]: Create **a** SonarQube Plugin

File: sonar-python-plugin/src/main/resources/static/documentation.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+* create a standard SonarQube plugin project
```
Comment (nicolas-harraudeau-sonarsource) [inline sonar-python-plugin/src/main/resources/static/documentation.md]: Missing dot at the end. Just to be coherent with the rest.

File: sonar-python-plugin/src/main/resources/static/documentation.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+* attach this plugin to the SonarQube Python analyzer through the `pom.xml`:
+  * add the dependency to the Python analyzer.
+  * add the following line in the sonar-packaging-maven-plugin configuration.
+  ```
+  <requirePlugins>python:2.0-SNAPSHOT</requirePlugin>
+  ```
+* implement the following extension points:
+  * [Plugin](http://javadocs.sonarsource.org/latest/apidocs/org/sonar/api/Plugin.html)
+  * [RulesDefinition](http://javadocs.sonarsource.org/latest/apidocs/org/sonar/api/server/rule/RulesDefinition.html) and [PythonCustomRuleRepository](https://github.com/SonarSource/sonar-python/blob/master/python-frontend/src/main/java/org/sonar/plugins/python/api/PythonCustomRuleRepository.java), which can be implemented by a single class, to declare your custom rules
```
Comment (nicolas-harraudeau-sonarsource) [inline sonar-python-plugin/src/main/resources/static/documentation.md]: missing dot at the end.

File: sonar-python-plugin/src/main/resources/static/documentation.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+To get started a sample plugin can be found here: [python-custom-rules](https://github.com/SonarSource/sonar-custom-rules-examples/tree/master/python-custom-rules).
+
+####  Implementation Details
+
+**Using `PythonCheckTree`**
+
+To explore a part of the AST, override a method from the PythonCheckTree. For example, if you want to explore "if statement" nodes, override [PythonCheckTree#visitIfStatement](https://github.com/SonarSource/sonar-python/blob/39b6126e9fdef42b93004cf6cc5818e861051334/python-frontend/src/main/java/org/sonar/plugins/python/api/tree/BaseTreeVisitor.java#L56) method that will be called each time an [ifStatement](https://github.com/SonarSource/sonar-python/blob/master/python-frontend/src/main/java/org/sonar/plugins/python/api/tree/IfStatement.java) node is encountered in the AST.
+
+![](/images/exclamation.svg) When overriding a visit method, you must call the super method in order to allow the visitor to visit the children the node.
```
Comment (nicolas-harraudeau-sonarsource) [inline sonar-python-plugin/src/main/resources/static/documentation.md]: "the children the node" => "the node's children" ?

File: sonar-python-plugin/src/main/resources/static/documentation.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  ```
+* implement the following extension points:
+  * [Plugin](http://javadocs.sonarsource.org/latest/apidocs/org/sonar/api/Plugin.html)
+  * [RulesDefinition](http://javadocs.sonarsource.org/latest/apidocs/org/sonar/api/server/rule/RulesDefinition.html) and [PythonCustomRuleRepository](https://github.com/SonarSource/sonar-python/blob/master/python-frontend/src/main/java/org/sonar/plugins/python/api/PythonCustomRuleRepository.java), which can be implemented by a single class, to declare your custom rules
+* declare the RulesDefinition as an extension in the Plugin extension point.
+
+#### Implement a Rule
+
+* create a class that will hold the implementation of the rule, it should:
+  * extend `PythonCheckTree` or `PythonSubscriptionCheck`
```
Comment (nicolas-harraudeau-sonarsource) [inline sonar-python-plugin/src/main/resources/static/documentation.md]: Dot at the end

File: sonar-python-plugin/src/main/resources/static/documentation.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  ```
+    <dependency>
+	  <groupId>org.sonarsource.python</groupId>
+	  <artifactId>python-checks-testkit</artifactId>
+	  <version>${project.version}</version>
+	  <scope>test</scope>
+    </dependency>
+  ```
+
+You should end each line with an issue with a comment in the following form:
```
Comment (nicolas-harraudeau-sonarsource) [inline sonar-python-plugin/src/main/resources/static/documentation.md]: "each line with an issue" => "each line having an issue" (double "with" is confusing)

**Pull request merged**